components/camel-soroush/src/main/java/org/apache/camel/component/soroushbot/utils/MultiQueueWithTopicThreadPool.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.soroushbot.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.RejectedExecutionException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * a simple thread pool that send each job to a thread based on the jobs topic,
+ */
+public class MultiQueueWithTopicThreadPool {
```

Comment:
Why not use either commons-pool, or use Java's ExecutorService which is also a thread pool

Reply:
Since we want that every message from the same user to be processed one by one, i.e. no 2 message from the same user execute concurrently, we create a new simple thread pool that let us select a thread by a topic. It guarantees that all tasks with the same `topic` execute in the same thread. We use `userIds` as the topic of each task.

Reply:
Ah okay, could you add the above to the javadoc so its a bit more clear why we are doing this, thanks.